src/main/java/hudson/remoting/PingThread.java

```diff
@@ -134,16 +134,16 @@ private void ping() throws IOException, InterruptedException {
      *      and provide a fallback behaviour to be backward compatible with earlier version of remoting library.
      */
     @Deprecated
-    protected abstract void onDead();
+    protected void onDead() {
+        onDead(new RuntimeException("No cause"));
```

Comment:
Sorry for missing the response. My idea was to put the null check logic to <code>protected void onDead(Throwable diagnosis)</code>, but it's need